baybe/surrogates/gaussian_process/kernel_factory.py

```
@@ -3,20 +3,118 @@
 from __future__ import annotations
 
 import gc
+from enum import Enum
 from typing import TYPE_CHECKING, Protocol
 
+import numpy as np
 from attrs import define, field
 from attrs.validators import instance_of
-from typing_extensions import override
+from typing_extensions import assert_never, override
 
 from baybe.kernels.base import Kernel
+from baybe.kernels.composite import AdditiveKernel, ProjectionKernel
 from baybe.searchspace import SearchSpace
 from baybe.serialization.mixin import SerialMixin
 
 if TYPE_CHECKING:
     from torch import Tensor
 
 
+def to_kernel_factory(x: Kernel | KernelFactory, /) -> KernelFactory:
+    """Wrap a kernel into a plain kernel factory (with factory passthrough)."""
+    return x.to_factory() if isinstance(x, Kernel) else x
```

No validation on the type of `x`? Asking as this can fail silently since *any* `x` that is not of type `Kernel` will simply be passed here.